### PR TITLE
Split CMakeLists.txt into sub-files

### PR DESCRIPTION
**Pull request by felipebalbi**
This will make it far easier to manage a growing project

### CODE REVIEW

**Comment by daviwil** [comment]
Thanks for sending this!  I intended to mention it on the stream yesterday but forgot about it, sorry about that!

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+include_directories(lib src)
+
+add_library(flux)
+target_link_libraries(flux PkgConfig::SDL2 PkgConfig::SDL2-Gfx
+  PkgConfig::SDL2-Image Threads::Threads)
+
+add_executable(flux-compose)
+target_link_libraries(flux-compose flux)
+
+add_subdirectory(lib)
```
Comment (daviwil) [inline CMakeLists.txt:25]: What's the difference between this and `include_directories`?

Comment (felipebalbi) [reply CMakeLists.txt:25]: `include_directories()` adds the directory to the include path (i.e. -Ilib). As for `add_subdirectory()` it adds the directory to as a build unit, i.e. it tells make to enter the subdirectory and look for other sources there.

**Comment by felipebalbi** [comment]
Rebased on top of `scripting` branch and made sure to split `test` directory too.

File: CMakeLists.txt (hunk 1 above)
Comment (daviwil) [reply CMakeLists.txt:25]: Ahhh, that makes sense.  Thanks a bunch!

**Pull request merged**